frontend/src/Components/Header/index.js

```diff
@@ -0,0 +1,83 @@
+import React, { useEffect, useState } from "react";
+import { Row, Col, Layout, Menu, Typography } from "antd";
+import { Link, useLocation } from "react-router-dom";
+import "./index.css";
+
+import UserStatus from "./UserStatus";
+
+const Header = () => {
+  const { Header } = Layout;
+  const { Title } = Typography;
+  const location = useLocation();
+  const [selected_keys, setSelected_keys] = useState(["0"]);
+
+  // Update menu selection based on current path
+  useEffect(() => {
+    const currentPath = location.pathname;
+    let first_dir = currentPath.substr(1).split("/")[0];
+    switch (String(first_dir)) {
+      case "":
+        setSelected_keys(["0"]);
+        break;
+      case "dashboard":
+        setSelected_keys(["1"]);
```

Comment:
I think you can use `NavLink` for this. https://reactrouter.com/web/api/NavLink

Reply:
I've tried to use NavLink but I couldn't find a way for it to style its parent. Surrounding the Menu.Item component by a NavLink, broke the Menu's styling.